src/edu/washington/escience/myria/operator/CSVFileScanFragment.java

```diff
@@ -179,30 +179,36 @@ public CSVFileScanFragment(
   @Override
   protected TupleBatch fetchNextReady() throws IOException, DbException {
     long lineNumberBegin = lineNumber;
+    boolean unfinishedEncapsulatedTokenRow = false;
 
     while ((buffer.numTuples() < TupleBatch.BATCH_SIZE) && !flagAsIncomplete) {
       lineNumber++;
       if (parser.isClosed()) {
         break;
       }
 
+      if (unfinishedEncapsulatedTokenRow) {
+        onLastRow = true;
+      }
+
       try {
         if (!onLastRow) {
           record = iterator.next();
         }
       } catch (Exception e) {
-        /*
-         * FIX ME: If we hit an exception for a malformed row (in case of quotes for example), we read more bytes by
-         * marking this as the final line
-         */
-        onLastRow = true;
       }
 
       try {
         if (!iterator.hasNext()) {
           onLastRow = true;
         }
       } catch (Exception e) {
```

Comment:
OK, I see now that the `IOException` is wrapped inside a `RuntimeException`. In that case, catch `Exception` and use Guava's `Throwables` to unwrap:
```
Throwables.getRootCause(e).getMessage());
```
(You might need to check `getMessage()` for `null` before calling string methods on it.) Then rethrow if the message doesn't match.